spring-cloud-huawei-router/src/main/java/com/huaweicloud/router/client/RouterClientAutoConfiguration.java

```diff
@@ -37,14 +39,24 @@
 @AutoConfigureAfter(LoadBalancerClientConfiguration.class)
 public class RouterClientAutoConfiguration {
   @Bean
-  public AbstractRouterDistributor<ServiceInstance, MicroserviceInstance> routerDistributor() {
-    return new SpringCloudRouterDistributor();
+  @ConditionalOnMissingBean(NacosRouterDistributor.class)
+  public AbstractRouterDistributor<ServiceInstance, CommonInstance> routerDistributorServicecomb(
+      MicroserviceInstanceService instance) {
+    return new SpringCloudRouterDistributor(instance);
+  }
+
+  @Bean
+  @ConditionalOnProperty(prefix = "spring.cloud.nacos.discovery", name = "service")
+  @Primary
+  public AbstractRouterDistributor<ServiceInstance, CommonInstance> routerDistributorNacos(
+      MicroserviceInstanceService instance) {
+    return new NacosRouterDistributor(instance);
   }
 
```

Comment:
Only one bean in router module. Can not have NacosRouterDistributor and SpringCloudRouterDistributor

Reply:
Modified